javascript/javascript1/week3/Homework/Exercise5.js

```diff
@@ -14,35 +15,57 @@ function addActivity(date, activity, duration) {
   activities.push(newActivity);
 }
 
-addActivity("23/7-18", "YouTube", 30);
-console.log(activities);
-
-// Show my status
-// Create a function called showStatus. This function should use the activities variable and return a string saying the following: "You have added 3 activities. They amount to 78 min. of usage"
-
 function calculateTotalDuration(total, activity) {
   return total + activity.duration;
 }
 
-function showStatus(activities) {
+function showStatus(date = new Date().toLocaleDateString()) {
+  const filteredActivities = activities.filter(
+    (activity) => activity.date === date
+  );
+
+  if (filteredActivities.length === 0) {
+    return `No activities found for ${date}`;
+  }
+
+  let totalActivities = filteredActivities.length;
+  let totalDuration = filteredActivities.reduce(calculateTotalDuration, 0);
+
+  return `You have added ${totalActivities} activities for ${date}. They amount to ${totalDuration} min. of usage`;
+}
+
+function mostTimeSpentActivity() {
   if (activities.length === 0) {
-    return "Add some activities before calling showStatus";
+    return "No activities recorded yet";
   }
 
-  let totalActivities = activities.length;
-  let totalDuration = activities.reduce(calculateTotalDuration, 0);
+  const activitiesByType = {};
+  activities.forEach((activity) => {
+    if (!activitiesByType[activity.activity]) {
+      activitiesByType[activity.activity] = 0;
+    }
+    activitiesByType[activity.activity] += activity.duration;
```

Comment:
You can also destructure arguments. This could be re-written as:

`activites.forEach(({ acivity, duration }) => ...`